src/flash/display/AVM1Movie.ts

```diff
@@ -45,27 +45,22 @@ module Shumway.AVM2.AS.flash.display {
     static classSymbols: string [] = null;
     static instanceSymbols: string [] = null;
 
-    constructor() {
+    constructor(content: MovieClip) {
       false && super();
       flash.display.DisplayObject.instanceConstructorNoInitialize.call(this);
       this._children = [];
-      this._constructed = false;
-    }
-
-    private _content: MovieClip;
-    private _constructed: boolean;
-
-    initializeContent(content: MovieClip): void {
-      release || assert(this._children.length === 0);
-      release || assert(!this._parent, "Must have content initialized before being added to stage");
       this._children[0] = this._content = content;
       // Pretend we're a DisplayObjectContainer and can have children. See comment at the top.
       content._setParent(<any>this, 0);
       this._setDirtyFlags(DisplayObjectFlags.DirtyChildren);
       this._invalidateFillAndLineBounds(true, true);
-      DisplayObject._advancableInstances.push(this);
+      display.DisplayObject._advancableInstances.push(this);
+      this._constructed = false;
     }
 
+    private _content: MovieClip;
+    private _constructed: boolean;
+
```

Comment:
This should be renamed to not be confused with DisplayObjectFlags.Constructed which we set to indicate that the constructor was already called on this object. Or maybe completely getting rid of this extra property and introducing DisplayObjectFlags.FirstFrameReady or DisplayObjectFlags.FirstFrameConstructed instead?